service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FaReportDocumentDao.java

```diff
@@ -172,18 +163,19 @@ public FaReportDocumentEntity findFaReportByRefIdAndRefScheme(String reportRefId
         for (FaReportDocumentEntity report : reports) {
             populateDeletingAndCancellationIds(report.getFishingActivities(), idsOfCancelledDeletedReports);
         }
+
         if (CollectionUtils.isNotEmpty(idsOfCancelledDeletedReports)) {
             return findReportsByIdsList(idsOfCancelledDeletedReports);
         }
-        return null;
+        return new ArrayList<>();
```

Comment:
Are you sure that no-one is checking to see if the returned value was null? I feel like there's a lot of that sort of code around... (also I'm getting déjà vu, I'm pretty sure I've written something exactly like this before in a review comment...)

Reply:
Only place that uses it does an empty check